### PR TITLE
feat: add getset method

### PR DESCRIPTION
**Pull request by yudhasubki**
Hi this PR is related to the issue
- add GetSet() to set a new value and return an old value

### CODE REVIEW

File: tx_btree.go [hunk 1]
```diff
@@ -207,6 +207,16 @@ func (tx *Tx) GetAll(bucket string) (values [][]byte, err error) {
 	return
 }
 
+func (tx *Tx) GetSet(bucket string, key, value []byte) (newValue, oldValue []byte, err error) {
```
Comment (bigboss2063) [inline tx_btree.go:210]: I think `GetSet` only need to return the old value.

Comment (bigboss2063) [reply tx_btree.go:210]: Just like Redis did.

File: tx_btree.go [hunk 2]
```diff
@@ -207,6 +207,15 @@ func (tx *Tx) GetAll(bucket string) (values [][]byte, err error) {
 	return
 }
 
+func (tx *Tx) GetSet(bucket string, key, value []byte) (oldValue []byte, err error) {
+	err = tx.update(bucket, key, func(b []byte) ([]byte, error) {
+		oldValue = b
+		return value, nil
+	})
+
+	return oldValue, err
```
Comment (bigboss2063) [inline tx_btree.go:216]: Maybe just `return oldValue, tx.update ...` will more elegant

Comment (yudhasubki) [reply tx_btree.go:216]: woh, looks good dude @bigboss2063.

**Review by bigboss2063 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**